components/automate-ui/src/app/pages/team-details/team-details.component.ts

```diff
@@ -237,4 +311,18 @@ export class TeamDetailsComponent implements OnInit, OnDestroy {
     // Drop the previous fragment and add the incoming fragment.
     this.router.navigate([this.url.split('#')[0]], { fragment: event.target.value });
   }
+
+  // updates whether the project was checked or unchecked
+  onProjectChecked(project: ProjectChecked): void {
+    this.projects[project.id] = project;
```

Comment:
Same thing mentioned on create-object-modal: Should do only what you are claiming:
```suggestion
    this.projects[project.id].checked = project.checked;
```
... and then you do not need the clarifying comment @bcmdarroch suggested. 😁